File: languages/go/entity/identityV2/proof_type.go
package identityV2

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

type ProofType int32

const (
	IntegrityProofType ProofType = iota
	SparseMtProofType  ProofType = iota
)

var (
	ProofTypeFromProto = map[proto.ProofType]ProofType{
		proto.ProofType_IntegrityProofType: IntegrityProofType,
		proto.ProofType_SparseMtProofType:  SparseMtProofType,
	}

	KeyTypeToProto = map[ProofType]proto.ProofType{
		IntegrityProofType: proto.ProofType_IntegrityProofType,
		SparseMtProofType:  proto.ProofType_SparseMtProofType,
	}
)

File: languages/php/Bloock/Entity/IdentityV2/ProofType.php
<?php

namespace Bloock\Entity\IdentityV2;

class ProofType
{
    const INTEGRITY_PROOF_TYPE = "INTEGRITY_PROOF_TYPE";
    const SPARSE_MT_PROOF_TYPE = "SPARSE_MT_PROOF_TYPE";

    public static function fromProto(\Bloock\ProofType $proofType): string
    {
        switch ($proofType) {
            case \Bloock\ProofType::IntegrityProofType:
                return ProofType::INTEGRITY_PROOF_TYPE;
            case \Bloock\ProofType::SparseMtProofType:
                return ProofType::SPARSE_MT_PROOF_TYPE;
            default:
                return null;
        }
    }

    public static function toProto(string $proofType): int
    {
        switch ($proofType) {
            case ProofType::INTEGRITY_PROOF_TYPE:
                return \Bloock\ProofType::IntegrityProofType;
            case ProofType::SPARSE_MT_PROOF_TYPE:
                return \Bloock\ProofType::SparseMtProofType;
            default:
                return null;
        }
    }
}
